name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DegiroPDFExtractor.java

```diff
@@ -159,257 +411,117 @@ private void addBankAccountTransactions()
-
-                                    t.removeUnit(unit);
-
-                                    long amountPlusDelta = unit.getAmount().getAmount() + delta;
-                                    long forexPlusDelta = BigDecimal.ONE
-                                                    .divide(unit.getExchangeRate(), 10, RoundingMode.HALF_DOWN)
-                                                    .multiply(BigDecimal.valueOf(amountPlusDelta))
-                                                    .setScale(0, RoundingMode.HALF_DOWN).longValue();
-
-                                    Unit newUnit = new Unit(unit.getType(),
-                                                    Money.of(unit.getAmount().getCurrencyCode(),
-                                                                    amountPlusDelta),
-                                                    Money.of(unit.getForex().getCurrencyCode(),
-                                                                    forexPlusDelta),
-                                                    unit.getExchangeRate());
-
-                                    t.addUnit(newUnit);
-                                }
-                            }
+    }
 
-                            return new TransactionItem(t);
-                        }));
-        
+    @SuppressWarnings("nls")
+    private FxChange getFxChangeFromContext(Map<String, String> fxContextMap, String date, String currencyCode, String amountFx)
+    {
+        String dateOnly = date.substring(0, date.indexOf(" ")); 
+        for (String key : fxContextMap.keySet())
+        {
+            String[] parts = key.split("_"); 
+            if (parts[0].equalsIgnoreCase("exchange")) 
+            {
+                // entry without valuta information
+                if (parts[2].equalsIgnoreCase(currencyCode) && (parts[1].equalsIgnoreCase(dateOnly)
+                                || fxContextMap.get(key).equalsIgnoreCase(amountFx)))
+                {
+                    return new FxChange(parts[1], Money.of(currencyCode, asAmount(fxContextMap.get(key))), parts[3],
+                                    parts[4]);
+                }
+                // entry with valuta
+                else if (parts[3].equalsIgnoreCase(currencyCode))
+                {
+                    //date matches
+                    if (parts[1].equalsIgnoreCase(dateOnly))
+                    {
+                        return new FxChange(parts[1], Money.of(currencyCode, asAmount(fxContextMap.get(key))), parts[4],
+                                        parts[5]);
+                    }
+                    //valuta matches
+                    else if (parts[2].equalsIgnoreCase(dateOnly))
+                    {
+                        return new FxChange(parts[2], Money.of(currencyCode, asAmount(fxContextMap.get(key))), parts[4],
+                                        parts[5]);
+                    }
```

Comment:
... dann ist dieses if-else egal, weil alle anderen Parameter für den Constructor dann identisch sind.

Was mir nicht klar ist: ist das data / valuta data dann komplett egal in diesem Fall? Oder müsste FxChange#getDate doch irgendwo geprüft werden?